### ISSUE TITLE
Incorrectly reporting import order in the presence of unused imports

### ISSUE THREAD

**Issue by smol-ninja** [protofire/solhint#675]
`solhint` incorrectly reports imports order warnings when there are unused imports, even if the order is correct.

### Steps to reproduce

Consider the following solidity code:

```solidity
// SPDX-License-Identifier: GPL-3.0-or-later
pragma solidity >=0.8.22;

import { AccessControl } from "@openzeppelin/contracts/access/AccessControl.sol";
import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
import { IERC721 } from "@openzeppelin/contracts/token/ERC721/IERC721.sol";

contract Aave {
    address public immutable MEV_PROTECTION_ADMIN;
}
```
#### Expected behaviour:

```shell
bun run solhint:check
$ solhint "{scripts/solidity,src,tests}/**/*.sol"

src/aave.sol
  4:10  warning  imported name AccessControl is not used     no-unused-import
  5:10  warning  imported name IERC20 is not used            no-unused-import
  6:10  warning  imported name IERC721 is not used           no-unused-import

✖ 6 problems (0 errors, 6 warnings)
```

#### Actual behaviour:

```shell
bun run solhint:check
$ solhint "{scripts/solidity,src,tests}/**/*.sol"

src/aave.sol
  2:1   warning  Wrong Import Order for {IERC721.sol}        imports-order
  2:1   warning  Wrong Import Order for {IERC20.sol}         imports-order
  2:1   warning  Wrong Import Order for {AccessControl.sol}  imports-order
  4:10  warning  imported name AccessControl is not used     no-unused-import
  5:10  warning  imported name IERC20 is not used            no-unused-import
  6:10  warning  imported name IERC721 is not used           no-unused-import

✖ 6 problems (0 errors, 6 warnings)
```
